GVFS/GVFS.Virtualization/Projection/GitIndexProjection.FolderData.cs

```diff
@@ -29,14 +31,22 @@ public void ResetData(LazyUTF8String name)
                 this.ChildEntries.Clear();
             }
 
-            public FolderData AddChildFolder(LazyUTF8String name)
+            public FileData AddChildFile(LazyUTF8String name, byte[] shaBytes)
             {
-                return this.ChildEntries.AddFolder(name);
+                return this.ChildEntries.AddFile(name, shaBytes);
             }
 
-            public FileData AddChildFile(LazyUTF8String name, byte[] shaBytes)
+            public void Include()
             {
-                return this.ChildEntries.AddFile(name, shaBytes);
+                this.IsIncluded = true;
+                for (int i = 0; i < this.ChildEntries.Count; i++)
+                {
+                    FolderData childFolder = this.ChildEntries[i] as FolderData;

```

Comment:
Minor/Nit: In GitIndexProject I saw cast to `FolderData` using this approach:

```
if (childEntry.IsFolder)
{
   FolderData folderData = (FolderData)childEntry;
```

Use the same approach here?